GraphicsView/include/CGAL/Basic_shaders.h

```diff
@@ -0,0 +1,277 @@
+  // rendering_mode == 0: draw inside only;
+  // rendering_mode == 1: draw outside only;
+  if (rendering_mode == (onPlane+1)/2) {
+    // discard other than the corresponding half when rendering
+    discard;
+  }
+
+  gl_FragColor = fColor;
+}
+)DELIM";
+
+const char vertex_source_clipping_plane[]=R"DELIM(
+#version 120
+attribute highp vec4 vertex;
+
+uniform highp mat4 vp_matrix;
+uniform highp mat4 m_matrix;
+
+void main(void)
+{
+  gl_Position = vp_matrix * m_matrix * vertex;
+}
+)DELIM";
+
+const char fragment_source_clipping_plane[]=R"DELIM(
+#version 120
+void main(void)
+{
+  gl_FragColor = vec4(0.0, 0.0, 0.0, 1.0);
+}
+)DELIM";
+
+//------------------------------------------------------------------------------
+//  compatibility shaders
+
+const char vertex_source_color_comp[]=R"DELIM(
+attribute highp vec4 vertex;
+attribute highp vec3 normal;
+attribute highp vec3 color;
+
+uniform highp mat4 mvp_matrix;
+uniform highp mat4 mv_matrix;
+
+varying highp vec4 fP;
+varying highp vec3 fN;
+varying highp vec4 fColor;
+
+uniform highp float point_size;
+
+void main(void)
+{
+  fP = mv_matrix * vertex;
+  highp mat3 mv_matrix_3;
+  mv_matrix_3[0] = mv_matrix[0].xyz;
+  mv_matrix_3[1] = mv_matrix[1].xyz;
+  mv_matrix_3[2] = mv_matrix[2].xyz;
+  fN = mv_matrix_3* normal;
+  fColor = vec4(color, 1.0);
+  gl_PointSize = point_size;
+
+  gl_Position = mvp_matrix * vertex;
+}
+)DELIM";
+
+const char fragment_source_color_comp[]=R"DELIM(
+varying highp vec4 fP;
+varying highp vec3 fN;
+varying highp vec4 fColor;
+
+uniform highp vec4 light_pos;
+uniform highp vec4 light_diff;
+uniform highp vec4 light_spec;
+uniform highp vec4 light_amb;
+uniform highp float spec_power ;
+
+void main(void)
+{
+  highp vec3 L = light_pos.xyz - fP.xyz;
+  highp vec3 V = -fP.xyz;
+
+  highp vec3 N = normalize(fN);
+  L = normalize(L);
+  V = normalize(V);
+
+  highp vec3 R = reflect(-L, N);
+  highp vec4 diffuse = max(dot(N,L), 0.0) * light_diff * fColor;
+  highp vec4 specular = pow(max(dot(R,V), 0.0), spec_power) * light_spec;
+
+  gl_FragColor = light_amb*fColor + diffuse;
+}
+)DELIM";
+
+const char vertex_source_p_l_comp[]=R"DELIM(
+attribute highp vec4 vertex;
+attribute highp vec3 color;
+
+uniform highp mat4 mvp_matrix;
+varying highp vec4 fColor;
+
+uniform highp float point_size;
+
+void main(void)
+{
+  gl_PointSize = point_size;
+  fColor = vec4(color, 1.0);
+  gl_Position = mvp_matrix * vertex;
+}
+)DELIM";
+
+const char fragment_source_p_l_comp[]=R"DELIM(
+varying highp vec4 fColor;
+void main(void)
+{
+  gl_FragColor = fColor;
+}
+)DELIM";
+
+/* const char vertex_source_clipping_plane_comp[]=R"DELIM(
+attribute highp vec4 vertex;
+
+uniform highp mat4 vp_matrix;
+uniform highp mat4 m_matrix;
+
+void main(void)
+{
+  gl_Position = vp_matrix * m_matrix * vertex;
+}
+)DELIM";
+
+const char fragment_source_clipping_plane_comp[]=R"DELIM(
+void main(void)
+{
+  gl_FragColor = vec4(0.0, 0.0, 0.0, 1.0);
+}
+)DELIM";
+*/
+
+}
+
+#endif // CGAL_BASIC_SHADERS_H
```

Comment:
I think the shaders should be in glsl version 150, like in #5336. Mac OS machines have some troubles with the version 120.

Reply:
Do you think it is enough to replace #version 120 by 150 ?

Reply:
Sadly no, you need to replace the `varying` and `attribute` specifiers too.
`attribute` becomes `in`, it is easy to replace, but varrying becomes `out` in vertex shaders and `in` in fragment shaders. 
You must also rpelace the pre-built `gl_FragColor` variable by a user-defind `out` variable in the fragment shaders. Just ask if you want me to take care of that, I did it quite a lot. 
You can also check the diff in #5336, it should be quite clear.

Reply:
Done in https://github.com/CGAL/cgal/pull/5406/commits/a7a3931f0ac67a730a7a063b53010066b5061526

But there are some errors, faces are now all black.
@maxGimeno Any idea where I am wrong please ?